src/Root.js

```diff
@@ -12,11 +14,22 @@ import { loadTranslations, checkOkapiSession } from './loginServices';
 import Stripes from './Stripes';
 import RootWithIntl from './RootWithIntl';
 
+import { modules } from 'stripes-config'; // eslint-disable-line
+
 class Root extends Component {
   constructor(...args) {
     super(...args);
     this.reducers = { ...initialReducers };
     this.epics = {};
+
+    for (const app of modules.app) {
+      if (window.location.pathname.startsWith(app.route) && app.queryResource) {
+        // This is not DRY, as it was expressed already in LocalResource is stripes-connect,
+        // And in MainNav.js in stripes-core. Both State Keys should be derived from a common mechanism.
+        this.queryResourceStateKey = `${app.dataKey ? `${app.dataKey}#` : ''}${_.snakeCase(app.module)}_${app.queryResource}`;
```

Comment:
I do agree that this should be factored out to a single location -- presumably a utility function in stripes-connect, which stripes-core can use. I don't think you should hold up merging the present PR while you do that, though. Come back and do it subsequently.